<html>
  <h1  align="center";> Diplomatura en Ciencia de Datos, Aprendizaje Automático y sus Aplicaciones
  </h1>
<hr>
  <h3 align="center";>
  Edición 2022
  </h3>
<hr>
  <h1 align="center";>
  Trabajo Práctico Entregable
  </h1>
  <h2 align="center";>
   Ejercicio 1 - Parte 1
  </h2>
</html>


## Profesora del Práctico:
- Fonseca, Rocio

## Integrantes [ Grupo 5 ]:
- Argüello, Leandro Ismael
- Corral, Diego
- Racciatti. Florencia 
- Rinaudo, Eleana
- Romero, Ernest

<html>
<hr>
  <h1  align="center";>
    Análisis Curación y Exploración de Datos
  </h1>
</html>


En esta notebook, vamos a cargar el conjunto de datos de [la compentencia Kaggle](https://www.kaggle.com/dansbecker/melbourne-housing-snapshot) sobre estimación de precios de ventas de propiedades en **Melbourne, Australia**.

Utilizaremos el conjunto de datos reducido producido por [DanB](https://www.kaggle.com/dansbecker). 

Hemos subido una copia a un servidor de la **Universidad Nacional de Córdoba** para facilitar su acceso remoto.

## Importación de librerías

Importamos la librerías que vamos a utilizar en el práctico.

In [ ]:
!pip install python-decouple;

In [ ]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

import seaborn as sns

sns.set_context("talk")

import missingno as msno

# from decouple import config
from sqlalchemy import create_engine, text


pd.set_option("display.max_columns", 1000)
pd.set_option("display.max_rows", 1000)


In [ ]:
import plotly

plotly.__version__
# Make sure it's 4.14.3


'5.5.0'

## Lectura del dataset

Cargamos el conjunto de datos de [la compentencia Kaggle](https://www.kaggle.com/dansbecker/melbourne-housing-snapshot) sobre estimación de precios de ventas de propiedades en **Melbourne, Australia**.

In [ ]:
melbourne_df = pd.read_csv(
    "https://cs.famaf.unc.edu.ar/~mteruel/datasets/diplodatos/melb_data.csv"
)
melbourne_df[:3]


Suburb          Address  Rooms Type      Price Method SellerG  \
0  Abbotsford     85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford  25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford     5 Charles St      3    h  1465000.0     SP  Biggin   

        Date  Distance  Postcode  Bedroom2  Bathroom  Car  Landsize  \
0  3/12/2016       2.5    3067.0       2.0       1.0  1.0     202.0   
1  4/02/2016       2.5    3067.0       2.0       1.0  0.0     156.0   
2  4/03/2017       2.5    3067.0       3.0       2.0  0.0     134.0   

   BuildingArea  YearBuilt CouncilArea  Lattitude  Longtitude  \
0           NaN        NaN       Yarra   -37.7996    144.9984   
1          79.0     1900.0       Yarra   -37.8079    144.9934   
2         150.0     1900.0       Yarra   -37.8093    144.9944   

              Regionname  Propertycount  
0  Northern Metropolitan         4019.0  
1  Northern Metropolitan         4019.0  
2  Northern Metropolitan         4019.0

---

## Ejercicio 1: **SQL**: 



1. Crear una base de datos en SQLite utilizando la libreria [SQLalchemy](https://stackoverflow.com/questions/2268050/execute-sql-from-file-in-sqlalchemy).
https://docs.sqlalchemy.org/en/14/core/engines.html#sqlite

Creamos la conexión a la base de datos.

In [ ]:
engine = create_engine("sqlite:///melbourne.sqlite3", echo=True)


---

2. Ingestar los datos provistos en 'https://cs.famaf.unc.edu.ar/~mteruel/datasets/diplodatos/melb_data.csv' en una tabla y el dataset generado en clase con datos de airbnb y sus precios por codigo postal en otra.



Generamos el Dataset de Melbourne a partir del `csv` provisto.

In [ ]:
melbourne_df = pd.read_csv(
    "https://cs.famaf.unc.edu.ar/~mteruel/datasets/diplodatos/melb_data.csv"
)
melbourne_df[:20]


Suburb              Address  Rooms Type      Price Method  SellerG  \
0   Abbotsford         85 Turner St      2    h  1480000.0      S   Biggin   
1   Abbotsford      25 Bloomburg St      2    h  1035000.0      S   Biggin   
2   Abbotsford         5 Charles St      3    h  1465000.0     SP   Biggin   
3   Abbotsford     40 Federation La      3    h   850000.0     PI   Biggin   
4   Abbotsford          55a Park St      4    h  1600000.0     VB   Nelson   
5   Abbotsford       129 Charles St      2    h   941000.0      S   Jellis   
6   Abbotsford         124 Yarra St      3    h  1876000.0      S   Nelson   
7   Abbotsford        98 Charles St      2    h  1636000.0      S   Nelson   
8   Abbotsford   6/241 Nicholson St      1    u   300000.0      S   Biggin   
9   Abbotsford        10 Valiant St      2    h  1097000.0      S   Biggin   
10  Abbotsford   411/8 Grosvenor St      2    u   700000.0     VB   Jellis   
11  Abbotsford      40 Nicholson St      3    h  1350000.0     VB   Nelson   
12  Abbotsford  123/56 Nicholson St      2    u   750000.0      S   Biggin   
13  Abbotsford        45 William St      2    h  1172500.0      S   Biggin   
14  Abbotsford   7/20 Abbotsford St      1    u   441000.0     SP     Greg   
15  Abbotsford        16 William St      2    h  1310000.0      S   Jellis   
16  Abbotsford          42 Henry St      3    h  1200000.0      S   Jellis   
17  Abbotsford          78 Yarra St      3    h  1176500.0      S   LITTLE   
18  Abbotsford     196 Nicholson St      3    h   955000.0      S  Collins   
19  Abbotsford        42 Valiant St      2    h   890000.0      S   Biggin   

          Date  Distance  Postcode  Bedroom2  Bathroom  Car  Landsize  \
0    3/12/2016       2.5    3067.0       2.0       1.0  1.0     202.0   
1    4/02/2016       2.5    3067.0       2.0       1.0  0.0     156.0   
2    4/03/2017       2.5    3067.0       3.0       2.0  0.0     134.0   
3    4/03/2017       2.5    3067.0       3.0       2.0  1.0      94.0   
4    4/06/2016       2.5    3067.0       3.0       1.0  2.0     120.0   
5    7/05/2016       2.5    3067.0       2.0       1.0  0.0     181.0   
6    7/05/2016       2.5    3067.0       4.0       2.0  0.0     245.0   
7    8/10/2016       2.5    3067.0       2.0       1.0  2.0     256.0   
8    8/10/2016       2.5    3067.0       1.0       1.0  1.0       0.0   
9    8/10/2016       2.5    3067.0       3.0       1.0  2.0     220.0   
10  12/11/2016       2.5    3067.0       2.0       2.0  1.0       0.0   
11  12/11/2016       2.5    3067.0       3.0       2.0  2.0     214.0   
12  12/11/2016       2.5    3067.0       2.0       2.0  1.0       0.0   
13  13/08/2016       2.5    3067.0       2.0       1.0  1.0     195.0   
14  14/05/2016       2.5    3067.0       1.0       1.0  1.0       0.0   
15  15/10/2016       2.5    3067.0       2.0       1.0  2.0     238.0   
16  16/07/2016       2.5    3067.0       3.0       2.0  1.0     113.0   
17  16/07/2016       2.5    3067.0       2.0       1.0  1.0     138.0   
18  17/09/2016       2.5    3067.0       3.0       1.0  0.0     183.0   
19  17/09/2016       2.5    3067.0       2.0       1.0  1.0     150.0   

    BuildingArea  YearBuilt CouncilArea  Lattitude  Longtitude  \
0            NaN        NaN       Yarra   -37.7996    144.9984   
1           79.0     1900.0       Yarra   -37.8079    144.9934   
2          150.0     1900.0       Yarra   -37.8093    144.9944   
3            NaN        NaN       Yarra   -37.7969    144.9969   
4          142.0     2014.0       Yarra   -37.8072    144.9941   
5            NaN        NaN       Yarra   -37.8041    144.9953   
6          210.0     1910.0       Yarra   -37.8024    144.9993   
7          107.0     1890.0       Yarra   -37.8060    144.9954   
8            NaN        NaN       Yarra   -37.8008    144.9973   
9           75.0     1900.0       Yarra   -37.8010    144.9989   
10           NaN        NaN       Yarra   -37.8110    145.0067   
11         190.0     2005.0       Yarra   -

---
Creamos una tabla llamada **Melbourne** y agregamos los records almacenados en el Dataframe dentro de dicha tabla.

In [ ]:
melbourne_df.to_sql("melbourne", con=engine, if_exists="replace")


2022-05-22 19:11:14,204 INFO sqlalchemy.engine.Engine PRAGMA main.table_info("melbourne")
2022-05-22 19:11:14,206 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:14,210 INFO sqlalchemy.engine.Engine PRAGMA main.table_info("melbourne")
2022-05-22 19:11:14,212 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:14,215 INFO sqlalchemy.engine.Engine SELECT name FROM sqlite_master WHERE type='table' ORDER BY name
2022-05-22 19:11:14,218 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:14,221 INFO sqlalchemy.engine.Engine PRAGMA main.table_info("melbourne")
2022-05-22 19:11:14,223 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:14,230 INFO sqlalchemy.engine.Engine SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
2022-05-22 19:11:14,232 INFO sqlalchemy.engine.Engine [raw sql] ('melbourne',)
2022-05-22 19:11:14,235 INFO sqlalchemy.engine.Engine PRAGMA main.foreign_key_list("m

---
 Creamos el Dataframe apartir del `csv` de **Airbnb** con las columnas de interés

In [ ]:
interesting_cols = [
    "id",
    "description",
    "neighborhood_overview",
    "street",
    "neighborhood",
    "city",
    "suburb",
    "state",
    "zipcode",
    "price",
    "weekly_price",
    "monthly_price",
    "latitude",
    "longitude",
]
airbnb_df = pd.read_csv(
    "https://cs.famaf.unc.edu.ar/~mteruel/datasets/diplodatos/cleansed_listings_dec18.csv",
    usecols=interesting_cols,
)


/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (35) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [ ]:
airbnb_df.tail(10)


id                                        description  \
22885  30591274                                                NaN   
22886  30591343  Clean; Comfortable; Quite; Beautiful; Convenie...   
22887  30591408  Located in the heart of Melbourne’s CBD, Upper...   
22888  30592088  Fantastic location close to public transport, ...   
22889  30592128  This is a idea for short stay backpacker who d...   
22890  30592332                       near city, near tram station   
22891  30592345  Close to: Melton Rail, Coles, Coffee Lounge an...   
22892  30592362  Hi, 我是一名中国留学生. 由于暑假原因决定出租我的房间. 我会留下我所使用的物品给我的租...   
22893  30592590  Immaculately presented and carefully maintaine...   
22894  30593623  An ENTIRE apartment unit ALL TO YOURSELF in Me...   

                                   neighborhood_overview  \
22885                                                NaN   
22886  The house is in a peaceful and convenient loca...   
22887                                                NaN   
22888  Tram to city, walking trails, great restaurant...   
22889                                                NaN   
22890                                                NaN   
22891                                                NaN   
22892                                                NaN   
22893  A plethora of Melbourne highlights are just a ...   
22894                                                NaN   

                              street               neighborhood        city  \
22885    Craigieburn, VIC, Australia                        NaN        Hume   
22886       Pakenham, VIC, Australia                        NaN    Cardinia   
22887      Melbourne, VIC, Australia  Central Business District   Melbourne   
22888       Hawthorn, VIC, Australia                   Hawthorn  Boroondara   
22889      Melbourne, VIC, Australia  Central Business District   Melbourne   
22890        Carlton, VIC, Australia                    Carlton   Melbourne   
22891   Melton South, VIC, Australia                        NaN      Melton   
22892  Glen Waverley, VIC, Australia                        NaN      Monash   
22893      Southbank, VIC, Australia                  Southbank   Melbourne   
22894        Carlton, VIC, Australia                    Carlton   Melbourne   

              suburb state zipcode   latitude   longitude  price  \
22885    Craigieburn   VIC  3064.0 -37.588371  144.901162     50   
22886       Pakenham   VIC  3810.0 -38.073001  145.441737     79   
22887      Melbourne   VIC  3000.0 -37.814443  144.952944    127   
22888       Hawthorn   VIC  3122.0 -37.830749  145.039288     84   
22889      Melbourne   VIC  3000.0 -37.819553  144.955837     30   
22890        Carlton   VIC  3053.0 -37.802415  144.962967     55   
22891   Melton South   VIC  3338.0 -37.705095  144.574474    129   
22892  Glen Waverley   VIC  3150.0 -37.875252  145.162757     53   
22893      Southbank   VIC  3006.0 -37.827795  144.954987    130   
22894        Carlton   VIC  3053.0 -37.805186  144.962399    179   

       weekly_price  monthly_price  
22885           NaN            NaN  
22886           NaN            NaN  
22887           NaN            NaN  
22888           NaN            NaN  
22889           NaN            NaN  
22890           NaN            NaN  
22891           NaN            NaN  
22892           NaN            NaN  
22893           NaN            NaN  
22894           NaN            NaN

Creamos una tabla llamada **Airbnb** y agregamos los **records** almacenados en el Dataframe dentro de dicha tabla.

In [ ]:
airbnb_df.to_sql("airbnb", con=engine, if_exists="replace")


2022-05-22 19:11:22,534 INFO sqlalchemy.engine.Engine PRAGMA main.table_info("airbnb")
2022-05-22 19:11:22,536 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:22,540 INFO sqlalchemy.engine.Engine PRAGMA main.table_info("airbnb")
2022-05-22 19:11:22,543 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:22,546 INFO sqlalchemy.engine.Engine SELECT name FROM sqlite_master WHERE type='table' ORDER BY name
2022-05-22 19:11:22,548 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:22,551 INFO sqlalchemy.engine.Engine PRAGMA main.table_info("airbnb")
2022-05-22 19:11:22,552 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:22,556 INFO sqlalchemy.engine.Engine SELECT sql FROM  (SELECT * FROM sqlite_master UNION ALL   SELECT * FROM sqlite_temp_master) WHERE name = ? AND type = 'table'
2022-05-22 19:11:22,560 INFO sqlalchemy.engine.Engine [raw sql] ('airbnb',)
2022-05-22 19:11:22,562 INFO sqlalchemy.engine.Engine PRAGMA main.foreign_key_list("airbnb")
2022

Declaramos las consultas que ejecutaremos en esta sección del práctico.

En **Melbourne**:

In [ ]:
query_melbourne_1 = "SELECT * FROM melbourne LIMIT 10"
query_melbourne_2 = "SELECT COUNT(1) FROM melbourne"
query_melbourne_3 = "SELECT DISTINCT Postcode FROM melbourne ORDER BY Postcode ASC"
query_melbourne_4 = "SELECT COUNT(CouncilArea), CouncilArea FROM melbourne GROUP BY CouncilArea ORDER BY CouncilArea ASC"
query_melbourne_5 = "SELECT COUNT(DISTINCT Postcode) FROM melbourne"
query_melbourne_6 = "SELECT COUNT(1) FROM melbourne WHERE Postcode IS NOT NULL"
query_melbourne_7 = "SELECT COUNT(1) FROM melbourne GROUP BY "


En **Airbnb**:

In [ ]:
airbnb_query_1 = "PRAGMA TABLE_INFO(airbnb)"
airbnb_query_2 = "SELECT COUNT(1) FROM airbnb"
airbnb_query_3 = "SELECT DISTINCT zipcode FROM airbnb ORDER BY zipcode ASC"
airbnb_query_4 = "SELECT * FROM airbnb ORDER BY city ASC LIMIT 10"
airbnb_query_5 = "SELECT SUM(CASE WHEN description IS NULL THEN 1 ELSE 0 END) AS Description, SUM(CASE WHEN neighborhood_overview IS NULL THEN 1 ELSE 0 END) AS neighborhood_overview, SUM(CASE WHEN street IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN neighborhood  IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN city IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN suburb IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN state  IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN zipcode IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN latitude IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN longitude IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN price IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN weekly_price IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN monthly_price  IS NULL THEN 1 ELSE 0 END) FROM airbnb"
airbnb_query_6 = "SELECT id, zipcode FROM airbnb WHERE LENGTH(zipcode) > 6 OR zipcode GLOB '*[a-zA-z]*'"
airbnb_query_7 = "UPDATE airbnb SET zipcode = '0.0' WHERE zipcode IS NULL"
airbnb_query_8 = (
    "SELECT id, SUBSTR(zipcode, 1, 4) FROM airbnb WHERE zipcode LIKE '%\n%'"
)
airbnb_query_9 = "SELECT id, zipcode FROM airbnb WHERE zipcode GLOB '*[a-zA-z]*'"
airbnb_query_10 = "DELETE FROM airbnb WHERE id="
airbnb_query_11 = "ALTER TABLE airbnb ADD COLUMN zipcode_float FLOAT NOT NULL DEFAULT 0"
airbnb_query_12 = "SELECT id, CAST(zipcode AS FLOAT) AS zipcode FROM airbnb"
airbnb_query_13 = "UPDATE airbnb SET zipcode_float = ? WHERE id = ?"
airbnb_query_14 = "SELECT COUNT(city), city FROM airbnb GROUP BY city ORDER BY city ASC"
airbnb_query_15 = "SELECT city, neighborhood, COUNT(*) FROM airbnb GROUP BY city, neighborhood ORDER BY city ASC"
airbnb_query_16 = "SELECT COUNT(DISTINCT zipcode_float) FROM airbnb"
airbnb_query_17 = "SELECT zipcode_float, AVG(price), AVG(weekly_price), AVG(monthly_price) FROM airbnb GROUP BY zipcode_float LIMIT 10"
airbnb_query_18 = "SELECT zipcode_float, AVG(price), COUNT(price), AVG(weekly_price), AVG(monthly_price) FROM airbnb GROUP BY zipcode_float LIMIT 10"
airbnb_query_19 = "SELECT price as Price, weekly_price AS Weekly, monthly_price AS Monthly FROM airbnb LIMIT 10"
airbnb_query_20 = "SELECT COUNT(1) FROM airbnb WHERE price IS NOT NULL"
airbnb_query_21 = "SELECT COUNT(1) FROM airbnb WHERE weekly_price IS NOT NULL"
airbnb_query_22 = "SELECT COUNT(1) FROM airbnb WHERE monthly_price IS NOT NULL"
airbnb_query_23 = (
    "SELECT price, neighborhood, suburb, zipcode FROM airbnb GROUP BY neighborhood"
)
airbnb_query_24 = "SELECT price, neighborhood, suburb, zipcode FROM airbnb GROUP BY zipcode ORDER BY  zipcode DESC"


En **Query en común**:

In [ ]:
common_query_1 = "SELECT COUNT(DISTINCT zipcode_float) FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode"
common_query_2 = "SELECT COUNT(Postcode) FROM melbourne WHERE Postcode IN (SELECT DISTINCT zipcode_float FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode)"
common_query_3 = "SELECT COUNT(zipcode_float) FROM airbnb WHERE zipcode IN (SELECT DISTINCT zipcode_float FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode)"


Se realizan a continuación algunas consultas para analizar los datos.

```sql
query_melbourne_1 = SELECT * FROM melbourne LIMIT 10
query_melbourne_2 = SELECT COUNT(1) FROM melbourne
query_melbourne_3 = SELECT DISTINCT Postcode FROM melbourne ORDER BY Postcode ASC
```

In [ ]:
queries = [query_melbourne_1, query_melbourne_2, query_melbourne_3]
with engine.connect() as con:
    for query in queries:
        rs = con.execute(query)
        print("\n\n", query)
        for row in rs:
            print(row)


2022-05-22 19:11:25,258 INFO sqlalchemy.engine.Engine SELECT * FROM melbourne LIMIT 10
2022-05-22 19:11:25,265 INFO sqlalchemy.engine.Engine [raw sql] ()


 SELECT * FROM melbourne LIMIT 10
(0, 'Abbotsford', '85 Turner St', 2, 'h', 1480000.0, 'S', 'Biggin', '3/12/2016', 2.5, 3067.0, 2.0, 1.0, 1.0, 202.0, None, None, 'Yarra', -37.7996, 144.9984, 'Northern Metropolitan', 4019.0)
(1, 'Abbotsford', '25 Bloomburg St', 2, 'h', 1035000.0, 'S', 'Biggin', '4/02/2016', 2.5, 3067.0, 2.0, 1.0, 0.0, 156.0, 79.0, 1900.0, 'Yarra', -37.8079, 144.9934, 'Northern Metropolitan', 4019.0)
(2, 'Abbotsford', '5 Charles St', 3, 'h', 1465000.0, 'SP', 'Biggin', '4/03/2017', 2.5, 3067.0, 3.0, 2.0, 0.0, 134.0, 150.0, 1900.0, 'Yarra', -37.8093, 144.9944, 'Northern Metropolitan', 4019.0)
(3, 'Abbotsford', '40 Federation La', 3, 'h', 850000.0, 'PI', 'Biggin', '4/03/2017', 2.5, 3067.0, 3.0, 2.0, 1.0, 94.0, None, None, 'Yarra', -37.7969, 144.9969, 'Northern Metropolitan', 4019.0)
(4, 'Abbotsford', '55a Park St', 4, 'h

A continuación realizamos las siguientes consultas para ver cantidad de registros y tipos de datos que tiene la tabla Airbnb.

  a. En la primer consulta se observa los **tipos de datos** de las columnas.  
  b. En la segunda consulta observamos que la tabla tiene **22895 registros**.  
  c. En la tercer consulta analizamos la columna `zipcode` y observamos que hay algunos valores erróneos en dicho campo (Ej: 3000\n3000, 3084\n\n3084, VIC 3161 y VIC).

Por otro lado, se observa en dicha columna que hay para iguales valores de `zipcode`, algunos almacenados como enteros y otros con decimales.  
d. En la consulta número cinco se observan que hay **142 registro** con valores `NULL` en la columna `zipcode`.

```sql
airbnb_query_1= PRAGMA TABLE_INFO(airbnb)
airbnb_query_2 = SELECT COUNT(1) FROM airbnb
airbnb_query_3 = SELECT DISTINCT zipcode FROM airbnb ORDER BY zipcode ASC
airbnb_query_4 = SELECT * FROM airbnb ORDER BY city ASC LIMIT 10
airbnb_query_5 = SELECT SUM(CASE WHEN description IS NULL THEN 1 ELSE 0 END) AS Description, SUM(CASE WHEN neighborhood_overview IS NULL THEN 1 ELSE 0 END) AS neighborhood_overview, SUM(CASE WHEN street IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN neighborhood  IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN city IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN suburb IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN state  IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN zipcode IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN latitude IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN longitude IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN price IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN weekly_price IS NULL THEN 1 ELSE 0 END), SUM(CASE WHEN monthly_price  IS NULL THEN 1 ELSE 0 END) FROM airbnb
```

In [ ]:
queries = [
    airbnb_query_1,
    airbnb_query_2,
    airbnb_query_3,
    airbnb_query_4,
    airbnb_query_5,
]
with engine.connect() as con:

    for query in queries:
        rs = con.execute(query)
        print("\n\n", query)
        for row in rs:
            print(row)


2022-05-22 19:11:25,376 INFO sqlalchemy.engine.Engine PRAGMA TABLE_INFO(airbnb)
2022-05-22 19:11:25,397 INFO sqlalchemy.engine.Engine [raw sql] ()


 PRAGMA TABLE_INFO(airbnb)
(0, 'index', 'BIGINT', 0, None, 0)
(1, 'id', 'BIGINT', 0, None, 0)
(2, 'description', 'TEXT', 0, None, 0)
(3, 'neighborhood_overview', 'TEXT', 0, None, 0)
(4, 'street', 'TEXT', 0, None, 0)
(5, 'neighborhood', 'TEXT', 0, None, 0)
(6, 'city', 'TEXT', 0, None, 0)
(7, 'suburb', 'TEXT', 0, None, 0)
(8, 'state', 'TEXT', 0, None, 0)
(9, 'zipcode', 'TEXT', 0, None, 0)
(10, 'latitude', 'FLOAT', 0, None, 0)
(11, 'longitude', 'FLOAT', 0, None, 0)
(12, 'price', 'BIGINT', 0, None, 0)
(13, 'weekly_price', 'FLOAT', 0, None, 0)
(14, 'monthly_price', 'FLOAT', 0, None, 0)
2022-05-22 19:11:25,399 INFO sqlalchemy.engine.Engine SELECT COUNT(1) FROM airbnb
2022-05-22 19:11:25,404 INFO sqlalchemy.engine.Engine [raw sql] ()


 SELECT COUNT(1) FROM airbnb
(22895,)
2022-05-22 19:11:25,410 INFO sqlalchemy.engine.Engine SELECT DISTINCT zipc

En la siguiente consulta se buscará convertir el tipo de datos de la columna `zipcode` de `TEXT` a `FLOAT`.

Para ello lo primero que hacemos en la consulta número seis, es visualizar los valores de `zipcode` cuya longitud es mayor a 6 y aquellos `zipcode` que contienen letras.

```sql
airbnb_query_6 = SELECT id, zipcode FROM airbnb WHERE LENGTH(zipcode) > 6 OR zipcode GLOB '*[a-zA-z]*'
```

In [ ]:
with engine.connect() as con:
    for query in queries:
        rs = con.execute(airbnb_query_6)
        print("\n\n", airbnb_query_6)
        for row in rs:
            print(row)


2022-05-22 19:11:25,920 INFO sqlalchemy.engine.Engine SELECT id, zipcode FROM airbnb WHERE LENGTH(zipcode) > 6 OR zipcode GLOB '*[a-zA-z]*'
2022-05-22 19:11:25,938 INFO sqlalchemy.engine.Engine [raw sql] ()


 SELECT id, zipcode FROM airbnb WHERE LENGTH(zipcode) > 6 OR zipcode GLOB '*[a-zA-z]*'
(1302086, 'VIC 3161')
(10095666, '3000\n3000')
(12459539, '3084\n\n3084')
(13788961, 'VIC')
2022-05-22 19:11:26,027 INFO sqlalchemy.engine.Engine SELECT id, zipcode FROM airbnb WHERE LENGTH(zipcode) > 6 OR zipcode GLOB '*[a-zA-z]*'
2022-05-22 19:11:26,034 INFO sqlalchemy.engine.Engine [raw sql] ()


 SELECT id, zipcode FROM airbnb WHERE LENGTH(zipcode) > 6 OR zipcode GLOB '*[a-zA-z]*'
(1302086, 'VIC 3161')
(10095666, '3000\n3000')
(12459539, '3084\n\n3084')
(13788961, 'VIC')
2022-05-22 19:11:26,093 INFO sqlalchemy.engine.Engine SELECT id, zipcode FROM airbnb WHERE LENGTH(zipcode) > 6 OR zipcode GLOB '*[a-zA-z]*'
2022-05-22 19:11:26,099 INFO sqlalchemy.engine.Engine [raw sql] ()


 SELECT id, zip

En la consulta número 7, le asignamos el valor 0.0 a todos los valores de `zipcode` iguales a `NULL`.

En la consulta número ocho, obtenemos los `zipcode` que tienen saltos de línea, los modificamos y actualizamos el valor en la tabla. 

Luego en la consulta número 9 obtenemos el id de los dos registros cuyo `zipcode` contienen letras y los eliminamos, debido a que solo son dos registros y no afectaría la integridad de los datos. 

```sql
airbnb_query_7 = UPDATE airbnb SET zipcode = '0.0' WHERE zipcode IS NULL
airbnb_query_8 = SELECT id, SUBSTR(zipcode, 1, 4) FROM airbnb WHERE zipcode LIKE '%\n%'
airbnb_query_9 = SELECT id, zipcode FROM airbnb WHERE zipcode GLOB '*[a-zA-z]*'
airbnb_query_10 = DELETE FROM airbnb WHERE id=
```

In [ ]:
with engine.connect() as con:
    con.execute(airbnb_query_7)
    rs = con.execute(airbnb_query_8)
    print("\n\n", airbnb_query_8)
    for row in rs:
        id = str(row[0])
        zipcode = str(row[1])
        query = "UPDATE airbnb SET zipcode = " + zipcode + " WHERE id=" + id
        con.execute(query)
        print("\n\n", query)

    rs = con.execute(airbnb_query_9)
    print("\n\n", airbnb_query_9)
    for row in rs:
        id = str(row[0])
        query = airbnb_query_10 + id
        con.execute(query)
        print("\n\n", query)


2022-05-22 19:11:26,297 INFO sqlalchemy.engine.Engine UPDATE airbnb SET zipcode = '0.0' WHERE zipcode IS NULL
2022-05-22 19:11:26,302 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:26,330 INFO sqlalchemy.engine.Engine COMMIT
2022-05-22 19:11:26,356 INFO sqlalchemy.engine.Engine SELECT id, SUBSTR(zipcode, 1, 4) FROM airbnb WHERE zipcode LIKE '%
%'
2022-05-22 19:11:26,362 INFO sqlalchemy.engine.Engine [raw sql] ()


 SELECT id, SUBSTR(zipcode, 1, 4) FROM airbnb WHERE zipcode LIKE '%
%'
2022-05-22 19:11:26,374 INFO sqlalchemy.engine.Engine UPDATE airbnb SET zipcode = 3000 WHERE id=10095666
2022-05-22 19:11:26,380 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:26,419 INFO sqlalchemy.engine.Engine COMMIT


 UPDATE airbnb SET zipcode = 3000 WHERE id=10095666
2022-05-22 19:11:26,471 INFO sqlalchemy.engine.Engine UPDATE airbnb SET zipcode = 3084 WHERE id=12459539
2022-05-22 19:11:26,479 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:26,513 INFO sqlalchemy

A continuación en la consulta número 11, creamos una nueva columna del tipo float llamada `zipcode_float`, donde almacenaremos el valor de `zipcode` convertido de `TEXT` a `float`. 

Luego en la consulta número 12, obtenemos todos los `zipcode`, convertimos de `TEXT` a `float`, para posteriormente en la consulta número 13 actualizar el valor en la nueva columna creada del tipo `FLOAT`.

```sql
airbnb_query_11 = ALTER TABLE airbnb ADD COLUMN zipcode_float FLOAT NOT NULL DEFAULT 0
airbnb_query_12 =SELECT id, CAST(zipcode AS FLOAT) AS zipcode FROM airbnb
airbnb_query_13 =UPDATE airbnb SET zipcode_float = ? WHERE id = ?
```

In [ ]:
with engine.connect() as con:
    con.execute(airbnb_query_11)
    rs = con.execute(airbnb_query_12)
    print(rs)
    result = []
    for row in rs:
        result.append([row[1], row[0]])

    con.execute(airbnb_query_13, result)


2022-05-22 19:11:26,706 INFO sqlalchemy.engine.Engine ALTER TABLE airbnb ADD COLUMN zipcode_float FLOAT NOT NULL DEFAULT 0
2022-05-22 19:11:26,721 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:26,732 INFO sqlalchemy.engine.Engine COMMIT
2022-05-22 19:11:26,742 INFO sqlalchemy.engine.Engine SELECT id, CAST(zipcode AS FLOAT) AS zipcode FROM airbnb
2022-05-22 19:11:26,747 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-22 19:11:27,081 INFO sqlalchemy.engine.Engine UPDATE airbnb SET zipcode_float = ? WHERE id = ?
2022-05-22 19:11:27,084 INFO sqlalchemy.engine.Engine [raw sql] [(3105.0, 9835), (3057.0, 10803), (3182.0, 12936), (3071.0, 15246), (3183.0, 16760), (3806.0, 38271), (3073.0, 41836), (3002.0, 43414)  ... displaying 10 of 22893 total bound parameter sets ...  (3006.0, 30592590), (3053.0, 30593623)]
2022-05-22 19:15:35,565 INFO sqlalchemy.engine.Engine COMMIT


> **Observación:**
>
> - Generamos el Dataset de **Melbourne** a partir del csv provisto y, además, creamos el Dataframe apartir del `csv` de **Airbnb** con las columnas de interés
> - Creamos una tabla llamada **Melbourne** y otra para **Airbnb** y agregamos los records almacenados en el Dataframe dentro de dichas tablas.
>- Declaramos algunas consultas para analizar los datos.
> - Realizamos consultas para ver cantidad de registros y tipos de datos que tiene la tabla Airbnb.
> - En la primer consulta, se observa los tipos de datos de las columnas.
> - En la segunda consulta, observamos que la tabla tiene **22895 registros**.
> - En la tercer consulta, analizamos la columna `zipcode` y observamos que hay algunos valores erróneos en dicho campo (Ej: 3000\n3000, 3084\n\n3084, VIC 3161 y VIC).
> - Por otro lado, se observa en dicha columna que hay para iguales valores de `zipcode`, algunos almacenados como enteros y otros con decimales.
> - Se observan que hay **142 registro** con valores `NULL` en la columna `zipcode`.
> - Convertimos el tipo de datos de la columna `zipcode` de `TEXT` a `FLOAT`. Para ello lo primero que hacemos en la consulta número seis, es visualizar los valores de `zipcode` cuya longitud es mayor a 6 y aquellos `zipcode` que contienen letras.
>- Le asignamos el valor 0.0 a todos los valores de `zipcode` iguales a NULL.
>- Obtenemos los `zipcode` que tienen saltos de línea, los modificamos y actualizamos el valor en la tabla.
>- Obtenemos el id de los dos registros cuyo `zipcode` contienen letras y los eliminamos, debido a que solo son dos registros y no afectaría la integridad de los datos. 
>- Creamos una nueva columna del tipo `float` llamada `zipcode_float`, donde almacenaremos el valor de `zipcode` convertido de `TEXT` a `float`.
>- Obtenemos todos los `zipcode`, convertimos de `TEXT` a `float`, para posteriormente actualizar el valor en la nueva columna creada del tipo `FLOAT`.

---
3. Implementar consultas en SQL que respondan con la siguiente información:

    - cantidad de registros totales por ciudad.
    - cantidad de registros totales por barrio y ciudad.

A continuación, en la consulta número 14, se obtienen la cantidad de registros totales por ciudad para la base de **Airnbnb** y en la consulta 4, la cantidad de registros totales por `CouncilArea` para la base de Melbourne. 

Debido a que esta última no posee la columna `city`. Se observa que no hay una relación directa y univoca entre `zipcode` y `city` o `CouncilArea`.

```sql
airbnb_query_14 = SELECT COUNT(city), city FROM airbnb GROUP BY city ORDER BY city ASC
query_melbourne_4 = SELECT COUNT(CouncilArea), CouncilArea FROM melbourne GROUP BY CouncilArea ORDER BY CouncilArea ASC
```

In [ ]:
with engine.connect() as con:
    rs = con.execute(airbnb_query_14)
    print(rs)

    for row in rs:
        print("\n", row)

with engine.connect() as con:
    rs = con.execute(query_melbourne_4)
    print(rs)
    for row in rs:
        print("\n", row)


with engine.connect() as con:
    rs = con.execute(
        "SELECT DISTINCT(CouncilArea) FROM melbourne WHERE Postcode = 3054"
    )
    print(rs)
    for row in rs:
        print("\n", row)

with engine.connect() as con:
    rs = con.execute("SELECT DISTINCT city FROM airbnb WHERE zipcode = 3054")
    print(rs)
    for row in rs:
        print("\n", row)


2022-05-22 19:15:35,637 INFO sqlalchemy.engine.Engine SELECT COUNT(city), city FROM airbnb GROUP BY city ORDER BY city ASC
2022-05-22 19:15:35,638 INFO sqlalchemy.engine.Engine [raw sql] ()

 (203, 'Banyule')

 (375, 'Bayside')

 (664, 'Boroondara')

 (108, 'Brimbank')

 (123, 'Cardinia')

 (153, 'Casey')

 (698, 'Darebin')

 (177, 'Frankston')

 (630, 'Glen Eira')

 (147, 'Greater Dandenong')

 (239, 'Hobsons Bay')

 (170, 'Hume')

 (309, 'Kingston')

 (175, 'Knox')

 (313, 'Manningham')

 (436, 'Maribyrnong')

 (115, 'Maroondah')

 (7368, 'Melbourne')

 (95, 'Melton')

 (571, 'Monash')

 (344, 'Moonee Valley')

 (967, 'Moreland')

 (88, 'Nillumbik')

 (2808, 'Port Phillip')

 (1620, 'Stonnington')

 (614, 'Whitehorse')

 (137, 'Whittlesea')

 (426, 'Wyndham')

 (2049, 'Yarra')

 (771, 'Yarra Ranges')
2022-05-22 19:15:35,698 INFO sqlalchemy.engine.Engine SELECT COUNT(CouncilArea), CouncilArea FROM melbourne GROUP BY CouncilArea ORDER BY CouncilArea ASC
2022-05-22 19:15:35,702 INFO sql

En la consulta número 15, se obtiene la cantidad de registros totales por barrio y por ciudad para la base de **Airbnb**.

```sql
airbnb_query_15 = SELECT city, neighborhood, COUNT(*) FROM airbnb GROUP BY city, neighborhood ORDER BY city ASC
```

In [ ]:
with engine.connect() as con:
    rs = con.execute(airbnb_query_15)
    print(rs)
    for row in rs:
        print("\n", row)


2022-05-22 19:15:35,816 INFO sqlalchemy.engine.Engine SELECT city, neighborhood, COUNT(*) FROM airbnb GROUP BY city, neighborhood ORDER BY city ASC
2022-05-22 19:15:35,817 INFO sqlalchemy.engine.Engine [raw sql] ()

 ('Banyule', None, 174)

 ('Banyule', 'Alphington', 16)

 ('Banyule', 'Preston', 8)

 ('Banyule', 'Thornbury', 5)

 ('Bayside', None, 170)

 ('Bayside', 'Bentleigh', 1)

 ('Bayside', 'Brighton', 203)

 ('Bayside', 'Elwood', 1)

 ('Boroondara', None, 30)

 ('Boroondara', 'Ashburton', 14)

 ('Boroondara', 'Balwyn', 70)

 ('Boroondara', 'Balwyn North', 40)

 ('Boroondara', 'Camberwell', 82)

 ('Boroondara', 'Canterbury', 24)

 ('Boroondara', 'Glen Iris', 32)

 ('Boroondara', 'Hawthorn', 261)

 ('Boroondara', 'Kew', 109)

 ('Boroondara', 'Malvern', 2)

 ('Brimbank', None, 108)

 ('Cardinia', None, 123)

 ('Casey', None, 153)

 ('Darebin', None, 112)

 ('Darebin', 'Alphington', 22)

 ('Darebin', 'Coburg', 4)

 ('Darebin', 'Northcote', 303)

 ('Darebin', 'Preston', 156)

 ('Dareb

> **Observación:**
>
> - En la consulta número 14, se obtienen la cantidad de **registros totales** por ciudad para la base de **Airnbnb** y en la consulta 4, la cantidad de **registros totales** por `CouncilArea` para la base de **Melbourne**.
> Debido a que esta última no posee la columna `city`. Se observa que no hay una relación directa y univoca entre `zipcode` y `city` o `CouncilArea`.
> - Obtenemos la cantidad de **registros totales por barrio y por ciudad** para la base de Airbnb.




---
4. Combinar los datasets de ambas tablas ingestadas utilizando el comando `JOIN` de `SQL`  para obtener un resultado similar a lo realizado con Pandas en clase. 

##### Se analizarán que hayan suficientes registros en común como para agregar información relevantes mediante la ejecución de las consultas 5 y 16 en ambas tablas respectivamente. Posteriormente se uniran los datos de ambas tablas mediante la consulta número 1. 

```sql
query_melbourne_5 = SELECT COUNT(DISTINCT Postcode) FROM melbourne
airbnb_query_16 = SELECT COUNT(DISTINCT zipcode_float) FROM airbnb
common_query_1 = SELECT COUNT(DISTINCT zipcode_float) FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode
```

In [ ]:
queries = [query_melbourne_5, airbnb_query_16, common_query_1]

with engine.connect() as con:

    for query in queries:
        rs = con.execute(query)
        print("\n\n", query)
        for row in rs:
            print(row)


2022-05-22 19:15:35,943 INFO sqlalchemy.engine.Engine SELECT COUNT(DISTINCT Postcode) FROM melbourne
2022-05-22 19:15:35,950 INFO sqlalchemy.engine.Engine [raw sql] ()


 SELECT COUNT(DISTINCT Postcode) FROM melbourne
(198,)
2022-05-22 19:15:35,964 INFO sqlalchemy.engine.Engine SELECT COUNT(DISTINCT zipcode_float) FROM airbnb
2022-05-22 19:15:35,973 INFO sqlalchemy.engine.Engine [raw sql] ()


 SELECT COUNT(DISTINCT zipcode_float) FROM airbnb
(248,)
2022-05-22 19:15:36,015 INFO sqlalchemy.engine.Engine SELECT COUNT(DISTINCT zipcode_float) FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode
2022-05-22 19:15:36,020 INFO sqlalchemy.engine.Engine [raw sql] ()


 SELECT COUNT(DISTINCT zipcode_float) FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode
(191,)


A continuación calcularemos lo siguiente:  

  a. La proporción de registros en la base de **Melbourne**, cuyo `Postcode` se encuentra en la intersección de ambas bases, mediante la consulta número 2.  
  b. La proporción de registros en la base de **Airbnb**, cuyo `zipcode` se encuentra en la intersección de ambas bases, mediante la consulta número 3. 

```sql
common_query_2 = SELECT COUNT(Postcode) FROM melbourne WHERE Postcode IN (SELECT DISTINCT zipcode_float FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode)
common_query_3 = SELECT COUNT(zipcode_float) FROM airbnb WHERE zipcode IN (SELECT DISTINCT zipcode_float FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode)
```

In [ ]:
with engine.connect() as con:

    rs = con.execute(common_query_2)
    for row in rs:
        a = row[0]

    rs = con.execute(query_melbourne_2)
    for row in rs:
        b = row[0]

    rs = con.execute(common_query_3)
    for row in rs:
        c = row[0]

    rs = con.execute(airbnb_query_2)
    for row in rs:
        d = row[0]

print("Records in Sales with corresponding zipcode form Melbourne: ", a / b)
print("Records in Sales with corresponding zipcode form Airbnb: ", c / d)


2022-05-24 22:30:02,365 INFO sqlalchemy.engine.Engine SELECT COUNT(Postcode) FROM melbourne WHERE Postcode IN (SELECT DISTINCT zipcode_float FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode)
2022-05-24 22:30:02,368 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-24 22:30:02,820 INFO sqlalchemy.engine.Engine SELECT COUNT(1) FROM melbourne
2022-05-24 22:30:02,832 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-24 22:30:02,838 INFO sqlalchemy.engine.Engine SELECT COUNT(zipcode_float) FROM airbnb WHERE zipcode IN (SELECT DISTINCT zipcode_float FROM airbnb A INNER JOIN melbourne M ON A.zipcode_float = M.Postcode)
2022-05-24 22:30:02,843 INFO sqlalchemy.engine.Engine [raw sql] ()
2022-05-24 22:30:03,292 INFO sqlalchemy.engine.Engine SELECT COUNT(1) FROM airbnb
2022-05-24 22:30:03,294 INFO sqlalchemy.engine.Engine [raw sql] ()
Records in Sales with corresponding zipcode form Melbourne:  0.9985272459499264
Records in Sales with corresponding zipcode form Airbnb:  0.93

A continuación combinamos ambas bases de datos. 

Mediante la aplicación del `INNER JOIN` lo que equivale a un **MERGE** en pandas, sin restringir la búsqueda.

In [ ]:
with engine.connect() as con:
    rs = con.execute(
        "SELECT COUNT(1) FROM airbnb INNER JOIN melbourne ON airbnb.zipcode_float = melbourne.Postcode"
    )

    for row in rs:
        print(row)


with engine.connect() as con:
    rs = con.execute(
        "SELECT COUNT(1) FROM airbnb LEFT JOIN melbourne ON airbnb.zipcode_float = melbourne.Postcode"
    )

    for row in rs:
        print(row)


2022-05-24 22:30:03,329 INFO sqlalchemy.engine.Engine SELECT COUNT(1) FROM airbnb INNER JOIN melbourne ON airbnb.zipcode_float = melbourne.Postcode
2022-05-24 22:30:03,336 INFO sqlalchemy.engine.Engine [raw sql] ()
(2139886,)
2022-05-24 22:30:03,491 INFO sqlalchemy.engine.Engine SELECT COUNT(1) FROM airbnb LEFT JOIN melbourne ON airbnb.zipcode_float = melbourne.Postcode
2022-05-24 22:30:03,493 INFO sqlalchemy.engine.Engine [raw sql] ()
(2141478,)


A continuación calcularemos los promedios de `price` por `zipcode`

a. Promedio de `price` por `zipcode`.  
b. Promedio de `weekly_price` por `zipcode`.  
c. Promedio de `monthly_price` por `zipcode`.  

```sql
airbnb_query_17 = SELECT zipcode_float, AVG(price), AVG(weekly_price), AVG(monthly_price) FROM airbnb GROUP BY zipcode_float LIMIT 10
airbnb_query_18 = SELECT zipcode_float, AVG(price), COUNT(price), AVG(weekly_price), AVG(monthly_price) FROM airbnb GROUP BY zipcode_float LIMIT 10
```

In [ ]:
with engine.connect() as con:
    rs = con.execute(airbnb_query_17)
    for row in rs:
        print(row)

    rs = con.execute(airbnb_query_18)
    for row in rs:
        print(row)


2022-05-24 22:30:03,658 INFO sqlalchemy.engine.Engine SELECT zipcode_float, AVG(price), AVG(weekly_price), AVG(monthly_price) FROM airbnb GROUP BY zipcode_float LIMIT 10
2022-05-24 22:30:03,661 INFO sqlalchemy.engine.Engine [raw sql] ()
(0.0, 160.96478873239437, 924.8181818181819, 3352.125)
(2010.0, 40.0, None, None)
(2134.0, 50.0, None, None)
(2582.0, 104.0, None, None)
(3000.0, 150.47001187648456, 918.7389558232932, 3407.204651162791)
(3001.0, 132.5, None, None)
(3002.0, 200.9796954314721, 956.3793103448276, 3712.714285714286)
(3003.0, 130.20599250936328, 760.2, 2629.5)
(3004.0, 158.27197802197801, 1061.1714285714286, 3759.925925925926)
(3006.0, 188.85173501577287, 1239.8272727272727, 4358.556701030928)



2022-05-24 22:30:03,714 INFO sqlalchemy.engine.Engine SELECT zipcode_float, AVG(price), COUNT(price), AVG(weekly_price), AVG(monthly_price) FROM airbnb GROUP BY zipcode_float LIMIT 10
2022-05-24 22:30:03,716 INFO sqlalchemy.engine.Engine [raw sql] ()
(0.0, 160.96478873239437, 142, 